extensions/components/com_redcore/site/helpers/translation.php

```diff
@@ -0,0 +1,167 @@
+<?php
```

Comment:
this looks like it is using many functions from admin helper RedcoreHelpersTranslation. You can move those functions to RTranslationHelper and use it from there as that is library that is always loaded.